engine/server/engine/centralized_logs/client_implementations/persistent_volume/log_file_creator/log_file_creator.go

```diff
@@ -0,0 +1,142 @@
+
+			serviceShortUuidFilePathStr := getFilepathStr(year, week, string(enclaveUuid), serviceShortUuidStr)
+			if err = creator.createSymlinkLogFile(serviceUuidFilePathStr, serviceShortUuidFilePathStr); err != nil {
+				return err
+			}
+			logrus.Tracef("Created symlinked log file: '%v'", serviceShortUuidFilePathStr)
+		}
+	}
+
+	return nil
+}
+
+func (creator *LogFileCreator) getEnclaveAndServiceInfo(ctx context.Context) (map[enclave.EnclaveUUID][]*service.ServiceRegistration, error) {
+	enclaveToServicesMap := map[enclave.EnclaveUUID][]*service.ServiceRegistration{}
+
+	enclaves, err := creator.kurtosisBackend.GetEnclaves(ctx, &enclave.EnclaveFilters{UUIDs: nil, Statuses: nil})
+	if err != nil {
+		return nil, stacktrace.Propagate(err, "An error occurred while trying to get all enclaves from kurtosis backend.")
+	}
+	for enclaveUuid := range enclaves {
+		var serviceRegistrations []*service.ServiceRegistration
+
+		enclaveServices, err := creator.kurtosisBackend.GetUserServices(ctx, enclaveUuid, &service.ServiceFilters{Names: nil, UUIDs: nil, Statuses: nil})
+		if err != nil {
+			return nil, stacktrace.Propagate(err, "An error occurred while trying to get user services for enclave '%v' from kurtosis backend.", enclaveUuid)
+		}
+		for _, serviceInfo := range enclaveServices {
+			serviceRegistrations = append(serviceRegistrations, serviceInfo.GetRegistration())
+		}
+
+		enclaveToServicesMap[enclaveUuid] = serviceRegistrations
+	}
+	return enclaveToServicesMap, nil
+}
+
+func (creator *LogFileCreator) createLogFileIdempotently(logFilePath string) error {
+	var err error
+	if _, err = creator.filesystem.Stat(logFilePath); os.IsNotExist(err) {
+		if _, err = creator.filesystem.Create(logFilePath); err != nil {
+			return stacktrace.Propagate(err, "An error occurred creating a log file path at '%v'", logFilePath)
+		}
+		logrus.Tracef("Created log file: '%v'", logFilePath)
+		return nil
+	}
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred checking if log file path at '%v' existed.", logFilePath)
+	}
+	return nil
+}
+
+func (creator *LogFileCreator) createSymlinkLogFile(targetLogFilePath, symlinkLogFilePath string) error {
+	// remove existing log files that could be storing logs at this path
+	if err := creator.filesystem.Remove(symlinkLogFilePath); err != nil {
+		return stacktrace.Propagate(err, "An error occurred attempting to remove an existing log file at the symlink file path '%v'.", symlinkLogFilePath)
+	}
+	// replace with symlink
+	if err := creator.filesystem.Symlink(targetLogFilePath, symlinkLogFilePath); err != nil {
+		return stacktrace.Propagate(err, "An error occurred creating a symlink file path '%v' for target file path '%v'.", targetLogFilePath, targetLogFilePath)
+	}
+	return nil
+}
+
+// creates a filepath of format /<filepath_base>/year/week/<enclave>/serviceIdentifier.<filetype>
+func getFilepathStr(year, week int, enclaveUuid, serviceIdentifier string) string {
```

Comment:
@tedim52 I think we should add a unit test for this guy alone to verify the format

Reply:
completely agree, if you don't mind I've added some unit tests in my logs `enclave rm` and `clean -a` that address this so gonna resolve this for now